### PR TITLE
Jsj stripe donate button

### LINKED ISSUE #31: Add Donation button
Two requests for this: https://github.com/rails-assets/rails-assets/issues/291#issuecomment-183279078

### PR DESCRIPTION
**Pull request by joshjordan**
Closes #31 (pending travis build)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,6 +7,8 @@ before_script:
   - cp config/database.sample.yml config/database.yml
   - psql -c 'create database rails_assets_test;' -U postgres
   - bundle exec rake db:migrate
+  - export DISPLAY=:99.0
+  - sh -e /etc/init.d/xvfb start
```
Comment (joshjordan) [inline .travis.yml]: Run `xvfb` for selenium. Stripe tests don't work well with capybara-webkit

File: app/assets/javascripts/application.coffee [hunk 2]
```diff
@@ -1,95 +1,16 @@
...
+#= require popeye
 #= require angular-semver-sort
+#= require js-routes
+#= require rails-assets
+#= require_directory .
 
-app = angular.module('app', ['semverSort'])
+app = angular.module('rails-assets')
 
-app.directive "ngHtml", ->
```
Comment (joshjordan) [inline app/assets/javascripts/application.coffee]: Moved this stuff as-is to `rails-assets.js`

File: app/assets/javascripts/rails-assets.js.coffee [hunk 3]
```diff
@@ -0,0 +1,92 @@
+app = angular.module('rails-assets', ['semverSort', 'ngNotificationsBar', 'ngAnimate', 'pathgather.popeye'])
```
Comment (joshjordan) [inline app/assets/javascripts/rails-assets.js.coffee]: This is a change - added 3 new dependencies. Notification bar and modal stuff.

File: app/assets/javascripts/stripe.js.coffee.erb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      if $scope.invokeStripe
+        $scope.invokeStripe = false
+        handler.open
+          name: 'Rails Assets'
+          description: 'Donation'
+          amount: $scope.amount * 100
+
+  $scope.next = ->
+    $scope.invokeStripe = true
+    $scope.$close()
```
Comment (joshjordan) [inline app/assets/javascripts/stripe.js.coffee.erb]: Oh. I think I can do this part smarter. Let me work on this.

File: app/controllers/components_controller.rb [hunk 5]
```diff
@@ -5,7 +5,7 @@ def index
     respond_to do |format|
       format.html { }
       format.json do
-        render(json: ComponentHelper.generate_components_json)
+        render nothing: true#render(json: ComponentHelper.generate_components_json)
```
Comment (joshjordan) [inline app/controllers/components_controller.rb]: Needs undoing.

File: app/controllers/donations_controller.rb [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  def log_donation
+    Donation.create!(
+      amount: @amount,
+      email: params[:token][:email],
+      client_ip: params[:token][:client_ip]
+    )
+  rescue => ex
+    Rails.logger.error "Failed to log donation: #{ex.message}"
+    Rails.logger.error ex.backtrace.join("\n")
+  end
```
Comment (joshjordan) [inline app/controllers/donations_controller.rb]: @hut8 can you open a ticket to Rollbar the application?

File: config/application.rb [hunk 7]
```diff
@@ -9,7 +9,7 @@
 
 require_relative '../app/workers/update_scheduler'
 
-module RailsAssets
+class RailsAssets
```
Comment (joshjordan) [inline config/application.rb]: I think this can be undone.

File: app/controllers/donations_controller.rb (hunk 6 above)
Comment (hut8) [inline app/controllers/donations_controller.rb]: done

**Pull request merged**